protocol/messenger.go

```diff
@@ -430,9 +433,27 @@ func (m *Messenger) Start() error {
 	m.watchConnectionChange()
 	m.watchExpiredEmojis()
 	m.watchIdentityImageChanges()
+	if err := m.cleanTopics(); err != nil {
+		return err
+	}
 	return nil
 }
 
+// cleanTopics remove any topic that does not have a Listen flag set
```

Comment:
This function takes filters from transport and instructs mailserver database to remove some. Looks like it doesn't work with anything messenger-specific. I would suggest not to make it a part of Messenger. Maybe it better fits Transport? Wdyt?

Reply:
I was thinking that eventually mailserver logic should be wrapped in messenger, as status-react should not be directly worrying about filters (it does now as we haven't fully migrated the code from status-react to status-go).
Essentially the flow should be for example:

1) Status-react creates a chat through messenger `CreateChat` endpoint
2) Messenger creates a chat, filter and mailserver topics and requests messages from mailserver

Instead currently we have

1) Status react creates a chat through messenger
2) Status-react ask messenger to create filters
3) Status-react saves the topics derived from filters to the database

So for now I agree it would make more sense to have it separate, but eventually mailserver logic should be handled by messenger, as it's dependent on creating chats etc (that's something I'd like to work next as we keep having issue with it)

What do you think?

Reply:
Thanks for explanation! With that in mind it absolutely has sense)